server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesRequest.java

```diff
@@ -88,6 +93,20 @@ public GetAliasesRequest indicesOptions(IndicesOptions indicesOptions) {
         return aliases;
     }
 
+    public GetAliasesRequest setAliases(String... aliases) {
```

Comment:
I don't think this is enough for transport client users. We are trying to fix the transport client behaviour, but effectively for this to work, transport client users need to migrate to the new setter, while the previous "setter" method is still around. How will users know that they have to move away from it? I would  rather move our own code (x-pack) to use a different method to replace aliases, and leave the original setter alone, but have it set the flag that we need. Would that be possible?

Reply:
I was afraid that this change would be much larger compared to what is currently done in the PR. But taking a second look at it, I think is is ok. I will rename the existing aliases(...) setter to replaceAliases(...)

Reply:
great!